halo2/src/circuit_builder.rs

```diff
@@ -152,9 +155,21 @@ impl<'a, T: FieldElement, F: PrimeField<Repr = [u8; 32]>> Circuit<F> for PowdrCi
         let advice = analyzed
             .committed_polys_in_source_order()
             .iter()
-            .flat_map(|(symbol, _)| symbol.array_elements())
-            // TODO(challenges): Set correct phase
-            .map(|(name, _)| (name.clone(), meta.advice_column_in(FirstPhase)))
+            .flat_map(|(symbol, _)| {
+                symbol
+                    .array_elements()
+                    .map(|(name, _)| (name, symbol.stage))
+            })
+            .map(|(name, stage)| {
+                let stage = stage.unwrap_or(0);
```

Comment:
I wonder if stage 0 should be the default instead of being an option?

Reply:
We could set those in the condenser - it should be stage-aware.

Reply:
Yes, that would make sense to me! I'd like to leave this for another PR though. It is also a bit weird that all symbols have stages, even though it is only relevant for witness columns.